src/vs/code/electron-sandbox/processExplorer/processExplorerMain.ts

```diff
@@ -374,22 +374,18 @@ class ProcessExplorer {
 		};
 
 		let matches = DEBUG_FLAGS_PATTERN.exec(item.cmd);
-		if (matches && matches.length >= 2) {
-			// attach via port
-			if (matches.length === 4 && matches[3]) {
-				config.port = parseInt(matches[3]);
-			}
-			config.protocol = matches[1] === 'debug' ? 'legacy' : 'inspector';
+		if (matches) {
+			config.port = parseInt(matches[1]);
 		} else {
 			// no port -> try to attach via pid (send SIGUSR1)
 			config.processId = String(item.pid);
 		}
 
 		// a debug-port=n or inspect-port=n overrides the port
 		matches = DEBUG_PORT_PATTERN.exec(item.cmd);
-		if (matches && matches.length === 3) {
+		if (matches) {
 			// override port
-			config.port = parseInt(matches[2]);
+			config.port = Number(matches[1]);
```

Comment:
Why not `parseInt`?

Reply:
`parseInt` can 'parse' a lot of stuff that isn't [just a number](https://stackoverflow.com/a/4090577/1125956). I prefer using `Number` if I know a string should only be numeric since it'll fail fast with a NaN.

Also I recall some benchmarks where Number is faster than parseInt, but that's a microoptimization 🤷‍♂️